app/models/provider_user.rb

```diff
@@ -37,7 +37,11 @@ def self.onboard!(dsi_user)
   end
 
   def full_name
-    "#{first_name} #{last_name}" if first_name.present? && last_name.present?
+    if first_name.present? && last_name.present?
+      "#{first_name} #{last_name}" if first_name.present? && last_name.present?
+    else
+      email_address
+    end
```

Comment:
The edit provider page sometimes shows no title because a provider user doesn’t have a name. Thinking in this case `full_name` could fall back to their email address. Firstly, is this a good idea, and if so, is this the right place in the code base to achieve this?

Reply:
This sounds more like a view concern, I wouldn't put it in the provider model, but it's not the worst place either! Perhaps you could create a `view_helper` for it, or make it part of a view component?

Reply:
Nice! Have made it a `view_helper`: `provider_user_name`.